src/dbus_api/blockdev.rs

```diff
@@ -177,52 +190,114 @@ fn set_user_info(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
+            }
+            _ => None,
+        })
+        .collect();
+
+    Ok(vec![return_message.append1(return_value)])
+}
+
+/// Perform an operation on a `BlockDev` object for a given
+/// DBus implicit argument that is a block device
+fn blockdev_operation<F, R>(
+    tree: &Tree<MTFn<TData>, TData>,
+    object_path: &Path<'static>,
+    closure: F,
+) -> Result<R, String>
 where
-    F: Fn(BlockDevTier, &dyn BlockDev) -> Result<R, MethodErr>,
+    F: Fn(BlockDevTier, &dyn BlockDev) -> Result<R, String>,
     R: dbus::arg::Append,
 {
-    let dbus_context = p.tree.get_data();
-    let object_path = p.path.get_name();
+    let dbus_context = tree.get_data();
 
-    let blockdev_path = p
-        .tree
+    let blockdev_path = tree
         .get(object_path)
         .expect("tree must contain implicit argument");
 
     let blockdev_data = blockdev_path
         .get_data()
         .as_ref()
-        .ok_or_else(|| MethodErr::failed(&format!("no data for object path {}", object_path)))?;
+        .ok_or_else(|| format!("no data for object path {}", object_path))?;
 
-    let pool_path = p.tree.get(&blockdev_data.parent).ok_or_else(|| {
-        MethodErr::failed(&format!(
-            "no path for parent object path {}",
-            &blockdev_data.parent
-        ))
-    })?;
+    let pool_path = tree
+        .get(&blockdev_data.parent)
+        .ok_or_else(|| format!("no path for parent object path {}", &blockdev_data.parent))?;
 
     let pool_uuid = pool_path
         .get_data()
         .as_ref()
-        .ok_or_else(|| MethodErr::failed(&format!("no data for object path {}", object_path)))?
+        .ok_or_else(|| format!("no data for object path {}", object_path))?
         .uuid;
 
     let engine = dbus_context.engine.borrow();
-    let (_, pool) = engine.get_pool(pool_uuid).ok_or_else(|| {
-        MethodErr::failed(&format!("no pool corresponding to uuid {}", &pool_uuid))
-    })?;
-    let (tier, blockdev) = pool.get_blockdev(blockdev_data.uuid).ok_or_else(|| {
-        MethodErr::failed(&format!("no blockdev with uuid {}", blockdev_data.uuid))
-    })?;
-    i.append(getter(tier, blockdev)?);
+    let (_, pool) = engine
+        .get_pool(pool_uuid)
+        .ok_or_else(|| format!("no pool corresponding to uuid {}", &pool_uuid))?;
+    let (tier, blockdev) = pool
+        .get_blockdev(blockdev_data.uuid)
+        .ok_or_else(|| format!("no blockdev with uuid {}", blockdev_data.uuid))?;
+    closure(tier, blockdev)
+}
+
+/// Get a blockdev property and place it on the D-Bus. The property is
+/// found by means of the getter method which takes a reference to a
+/// blockdev and obtains the property from the blockdev.
+fn get_blockdev_property<F, R>(
+    i: &mut IterAppend,
+    p: &PropInfo<MTFn<TData>, TData>,
+    getter: F,
+) -> Result<(), MethodErr>
+where
+    F: Fn(BlockDevTier, &dyn BlockDev) -> Result<R, String>,
+    R: dbus::arg::Append,
+{
+    let object_path = p.path.get_name();
```

Comment:
Minor: I like to have as few identifiers as possible, so I avoid them if they're only used once; here you could remove the ```object_path``` identifier altogether and probably get nicer indentation.

NOTE: Applies to ```filesystem.rs``` and ```pool.rs``` as well.

Reply:
Resolved.